docs/rfc/rfc-022-semi-permanent-testnet.md

```diff
@@ -0,0 +1,202 @@
+These goals would generally be beyond the one-year plan,
+and therefore they would not be part of initial planning.
+
+- There is a centralized dashboard to get a quick overview of all testnets,
+  or at least one centralized dashboard per testnet,
+  showing TBD basic information
+- Testnets include cloud spot instances which periodically and abruptly join and leave the network
+- The testnets are a heterogeneous mixture of straight VMs and Docker containers,
+  thereby more closely representing production blockchains
+- Testnets have some manner of continuous profiling,
+  so that we can produce an apples-to-apples comparison of CPU/memory cost of particular operations
+
+### Testnet non-goals
+
+There are some things we are explicitly not trying to achieve with long-lived testnets:
+
+- The testnets will NOT be maintained with any kind of on-call schedule;
+  although all engineers directly working on Tendermint will be expected to contribute to
+  the ongoing monitoring and maintenance of the testnets,
+  they have no obligations to maintain the testnet in any fashion after working hours
+- The testnets will NOT be used to gate pull requests;
+  that responsibility belongs to unit tests, end-to-end tests, and integration tests
+- Similarly, the testnet will NOT be used to automate any changes back into Tendermint source code;
+  we will not automatically create a revert commit due to a failed rollout, for instance
+- The testnets are NOT intended to have participation from machines outside of the
+  Tendermint engineering team's control, as the Tendermint engineers are expected
+  to have full access to any instance where they may need to debug an issue
+- While there will certainly be individuals within the Tendermint engineering team
+  who will continue to build out their individual "devops" skills to produce
+  the infrastructure for the testnet, it is NOT a goal that every Tendermint engineer
+  is even _familiar_ with the tech stack involved, whether it is Ansible, Terraform,
+  Kubernetes, etc.
+  As a rule of thumb, all engineers should be able to get shell access on any given instance
+  and should have access to the instance's logs.
+  Little if any further operational skills will be expected.
+- The testnets are not intended to be _created_ for one-off experiments.
+  While there is nothing wrong with an engineer directly interacting with a testnet
+  to try something out,
+  a testnet comes with a considerable amount of "baggage", so end-to-end or integration tests
+  are closer to the intent for "trying something to see what happens".
+- Likewise, the purpose of the testnet is not to run specific "tests" per se,
+  but rather to demonstrate that Tendermint blockchains as a whole can be stable
+  under a production load.
+  Of course we will inject faults periodically, but the intent is to observe and prove that
+  the testnet is resilient to those faults.
```

Comment:
From this I derive that we can inject malicious/faulty behavior, but this behavior should not in principle halt or produce irrecoverable errors in the chain?

Reply:
I've thought about this for a few minutes and I'm not sure how to add it to the document.

If you knew that running a particular command did something strange on a 3-node testnet, and you were curious what effect it had on a 100-node testnet: I think it would be okay to run that against the main testnet, even with an uncertain risk that it would halt the testnet.

But if you were certain that you could halt the main testnet with a single command, then I would recommend that you would not run it -- unless perhaps you were going to confirm that the next software update was capable of recovering from said halt.

This is a kind of fuzzy area that I think is fine to omit from the document for now.